### DISCUSSION TITLE
MessageBox text

### DISCUSSION THREAD

**Discussion by AlquistArj** [FlaUI/FlaUI#546] [General]
Is there any way I can read the text in a popup MessageBox? I can see, and read the Title and the text in the buttons but strangely cannot seem to access the text (or icon) of the message using FlaUInspect V1.3.0.

**Comment by jwu-dharmo** [comment]
Have you tried GetModal()?  if your MessageBox is not modal, you will have to use the Inspect Object tool from the Windows kit to find the AutomationId, then you will be able to get the window.  Also, knowing which parent window that launches the MsgBox will help A LOT!  :)  Good luck!

**Comment by AlquistArj** [comment]
Let me show an example... Here I can see the main OK button but not the text above. It's the text I want to read to find out what the error is.
![Untitled](https://user-images.githubusercontent.com/11150083/214411344-0026c8c8-c287-4da3-9e20-03764010693d.png)

**Comment by LevYas** [comment]
Since there are no better answers... If a test case fails, I make a screenshot and post it as an artifact. I use xUnit.Context:
```
internal class ScreenshotSaver
{
    public static void Save(Context context)
    {
        try
        {
            string pathToDir = Path.Combine(EnvConstants.PathToTestOutput, "Pics");
            if (!Directory.Exists(pathToDir))
                Directory.CreateDirectory(pathToDir);

            // Use third-party context because xUnit will release it only in v3 https://github.com/xunit/xunit/issues/621
            ITestMethod method = context.Test.TestCase.TestMethod;
            string name = $"{method.TestClass.Class.Name}_{method.Method.Name}".Replace('.', '_');

            string targetPath = Path.Combine(pathToDir, $"{name}.png");

            FlaUI.Core.Capturing.Capture.Screen().ToFile(targetPath);
        }
        catch (Exception e)
        {
            Console.WriteLine($"Failed to write the screenshot because of exception {e}");
        }
    }
}

**Reply by LevYas** [reply]
You can also recognize the text on the screen withing bounding box like this project does https://github.com/TheJoeFin/Text-Grab
